### ISSUE TITLE
The segmentation fault(core dump) error after hundreds of training iterations 

### ISSUE THREAD

**Issue by sailor-z** [zjhthu/OANet#7]
Hi,
Thanks for the amazing work.
I'm trying to train your OANet and my own method on YFCC100M dataset. But there is a segmentation fault(core dump) error after hundreds of or thousands of training iterations. Have you ever got the similar error?

**Comment by zjhthu** [comment]
Is there any other information given when it stops? Sometimes it will go wrong in `torch.symeig` but pytorch will print something.

**Comment by sailor-z** [comment]
![image](https://user-images.githubusercontent.com/26895204/77322830-4a066b00-6d4f-11ea-975d-72413ac364d5.png)
There is no other information.  It seems like a memory error.

**Comment by zjhthu** [comment]
It's quite weird. I should never encounter this problem recently. 
Do you use the same code and data? What is your pytorch version?

**Comment by sailor-z** [comment]
Yes, I used the same code and data and encountered this problem. So I've integrated your dataloader into my training code, but I still encounter the same problem. My pytorch version is 1.4.

**Comment by zjhthu** [comment]
One possible reason I can think of is that there are some strange data that cause unexpected errors. To debug this, you can set `shuffle=False` and see whether the code dumps at the same data.

**Comment by sailor-z** [comment]
I have successfully trained one epoch so I don't think it is caused by strange data. I have googled it and it might be because one is working with a lot of data and the RAM is full, or the Python version is improper. Could you please tell me your Python version and the RAM size of your computer?

**Comment by zjhthu** [comment]
124G and python 3.5. You can monitor the memory usage to see whether it is because of the memory.

**Comment by sailor-z** [comment]
I have monitored the memory usage. Memory leak will occur if I set num_workers > 0. Someone encounters the similar error: https://github.com/pytorch/pytorch/issues/13246 and I need to further debug to figure it out.

**Comment by zjhthu** [comment]
I retrained several models yesterday, but no memory leaks were found...

**Comment by sailor-z** [comment]
I've retrained you method and successfully reproduced the results reported in your paper. Thanks for your help.

**Comment by zjhthu** [comment]
So, did you find the reason for the segmentation fault?
